Add `Active` and `S3CanonicalUserID` to `UserProperties`

piepmatz opened the pull request:
The `Active` field was forgotten in the API's swagger document.
The API team confirmed that the field can be relied on, though.
A bug was filed regarding the incomplete swagger doc.

The `S3CanonicalUserID` field was just missing in the SDK.

avorima commented on usermanagment.go:
```diff
@@ -73,13 +73,15 @@ type User struct {
+	ForceSecAuth      bool   `json:"forceSecAuth,omitempty"`
+	SecAuthActive     bool   `json:"secAuthActive,omitempty"`
+	Active            bool   `json:"active,omitempty"`
+	S3CanonicalUserID string `json:"s3CanonicalUserId,omitempty"`
```
I assume this field was also forgotten in the swagger, but can be relied upon?

piepmatz replied:
No, it's in swagger, I didn't touch this field apart from indentation changes.

avorima replied:
OK. But it is a new field in the struct, so could you please mention it in the PR title or comment?

piepmatz replied:
You are right. I added this field. I can't remember. I was pretty sure I only added `Active`. But git proves me wrong. So yeah, I'll also adapt the unit test.

avorima reviewed (approved):
LGTM

piepmatz commented:
@pennycoders PTAL

pennycoders commented:
Hi,

Can you please update the README.md file as well please?

Best Regards,

Alex

piepmatz commented:
Hi Alex,

I updated the README.

I am really unhappy about the README in general. There are so many mistakes, so much inconsistencies, so much incomplete things.
I tried to improve things a bit for the `User`-related stuff, but I am still far from happy about all this.

Just as example for the `UserProperties` I touched. Some fields can only be sent in `POST` but must not be sent in `PUT`. A single generic table as in `User Resource Object` does not reflect that.
Moreover, using `bool` for `Administrator`, `ForceSecAuth` and `SecAuthActive` is a bug. I didn't change it to `*bool` as this would be a breaking change and is actually out of scope for this PR.

Back to the README in general. Who's the SDK's audience? Developers. Personally, I have never looked into the README, not a single time. My IDE lists the client's methods, makes jumping to the definitions easy. Right there could be doc strings, that would be helpful. And if they were there, I wouldn't have forgotten writing something about the fields I added. But the README is far away, so forgetting about it is easy.
For more general things I consult https://devops.ionos.com/api/cloud/v5/ or, as it's also far from perfect, the swagger document. Which also has lots of problems, e.g. it doesn't distinguish between the `User` payloads for `POST` and `PUT` and their differences. That's why I am also a bit sceptical regarding the new automatically created Swagger-based SDK you work on.

You guys maintain several SDKs. It must be a PITA to document each and every parameter for every endpoint. Of course this must be done, but isn't it enough to have it once at https://devops.ionos.com/api/cloud/v5/? Is there benefit of more or less copying it to all the SDK's READMEs? It just super easy to have outdated READMEs all the time.

I think the README is great for getting started and some examples, but I doubt it's worth all the work required to make it a proper documentation.

Sorry for the rant. Editing the README just felt completely useless.

pennycoders commented:
Hi Matt,

I understand what you are saying, and I agree, the README.md should have been updated, but that is the state we have found it in. The most important part from the README are indeed the samples. Thank you for updating the README.md. The SDK is being rewritten, so that everything (SDK's, documentation) is being auto-generated based on the Swagger specification - that should solve all documentation-related inconsistencies as well as the SDK-related ones.

Best Regards,

Alex
